docs/src/pldm.md

```diff
@@ -0,0 +1,268 @@
+///   Processes an incoming request message.
+///   - `payload`: A mutable reference to the payload data to be processed.
+///   - Returns a `Result` containing the size of the processed data or a `MessageHandlerError`.
+///
+/// - `process_response(&self, payload: &mut [u8]) -> Result<usize, MessageHandlerError>`:
+///   Processes an incoming response message.
+///   - `payload`: A mutable reference to the payload data to be processed.
+///   - Returns a `Result` containing the size of the processed data or a `MessageHandlerError`.
+///
+/// # Errors
+///
+/// Both methods return a `Result` which, on failure, contains a `MessageHandlerError`.
+pub trait MessageResponder: Send + Sync {
+    async fn process_request(&self, payload: &mut [u8]) -> Result<usize, MessageHandlerError>;
+    async fn process_response(&self, payload: &mut [u8]) -> Result<usize, MessageHandlerError>;
+}
+
+/// Trait representing a message requester that can generate requests asynchronously.
+pub trait MessageRequester: Send + Sync {
+    /// Asynchronously generates a request with the given payload.
+    ///
+    /// # Arguments
+    ///
+    /// * `payload` - A mutable reference to a byte slice that will be used to generate the request.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing the size of the generated request on success, or a `MessageHandlerError` on failure.
+    async fn generate_request(&self, payload: &mut [u8]) -> Result<usize, MessageHandlerError>;
+}
+pub struct MessageHandlerError(pub NonZeroU32);
+
+
+/// Trait representing a command handler that can execute commands asynchronously.
+///
+/// # Required Methods
+///
+/// - `execute(&self, payload: &mut [u8], context: &mut PldmContext) -> usize`:
+///   Executes a command with the given payload and context.
+///   - `payload`: A mutable reference to the payload data to be processed.
+///   - `context`: A mutable reference to the PLDM context.
+///   - Returns the size of the processed data.
+///
+/// - `send_notification()`:
+///   Sends a notification to other modules or APIs asynchronously.
+///
+/// - `receive_notification()`:
+///   Receives a notification from other modules or APIs asynchronously.
+pub trait CommandHandler: Send + Sync {
+    async fn execute(&self, payload: &mut [u8], context: &mut PldmContext) -> usize;
+    async fn send_notification(&self);
+    async fn receive_notification(&self);
+}
+
+/// Represents a PLDM Command Interface Responder.
+///
+/// This struct is responsible for handling the PLDM command interface
+/// by maintaining a collection of command handlers and the PLDM context.
+///
+/// # Fields
+///
+/// * `handlers` - A hashmap that maps command codes to their respective command handlers.
+/// * `context` - The PLDM context used for processing commands.
+/// * `other fields` - Additional fields required for the command interface responder.
+pub struct CmdInterfaceResponder {
+    handlers: HashMap<u8, Box<dyn CommandHandler>>,
```

Comment:
I think we're not going to have `HashMap` and `Box` available, since they require allocations. Instead, we should probably use the `LinkList` structures that some of the virtualizers and muxes use: https://github.com/tock/tock/blob/master/capsules/core/src/virtualizers/virtual_alarm.rs#L46